pkg/api/vlabs/validate.go

```diff
@@ -263,6 +263,9 @@ func (a *Properties) validateNetworkPolicy() error {
 
 	switch a.OrchestratorProfile.OrchestratorType {
 	case Kubernetes:
+		if a.OrchestratorProfile.KubernetesConfig == nil {
+			return nil
+		}
```

Comment:
this is returning an error as this is validation. so returning nil means this is a valid value. I'll skip the rest of this case which will leave networkPolicy as "" which is how this method behaves today.